.cbmc-batch/jobs/aws_add_size_checked/aws_add_size_checked_harness.c

```diff
@@ -2,48 +2,30 @@
 #include <proof_helpers/make_common_data_structures.h>
 
 /**
- * Adds a + b and returns the result in *r. If the result
- * overflows, returns AWS_OP_ERR; otherwise returns AWS_OP_SUCCESS.
- */
-AWS_STATIC_IMPL int __aws_add_u64_checked(uint64_t a, uint64_t b, uint64_t *r) {
-    uint64_t x = a + b;
-    *r = x;
-    if (x < a) {
-        return aws_raise_error(AWS_ERROR_OVERFLOW_DETECTED);
-    }
-    return AWS_OP_SUCCESS;
-}
-
-/**
- * Adds a + b and returns the result in *r. If the result
- * overflows, returns AWS_OP_ERR; otherwise returns AWS_OP_SUCCESS.
- */
-AWS_STATIC_IMPL int __aws_add_u32_checked(uint32_t a, uint32_t b, uint32_t *r) {
-    uint32_t x = a + b;
-    *r = x;
-    if (x < a) {
-        return aws_raise_error(AWS_ERROR_OVERFLOW_DETECTED);
-    }
-    return AWS_OP_SUCCESS;
-}
-
-/**
+ * Coverage: 1.00 (29 lines out of 29 statically-reachable lines in 5 functions reached)
  * Runtime: 0m2.025s
+ *
+ * Assumptions:
+ *     - given 2 non-deterministics unsigned integers
+ *
+ * Assertions:
+ *     - r does not overflow, if aws_add_u32_checked or
+ *       aws_add_u64_checked functions return AWS_OP_SUCCESS
  */
 void aws_add_size_checked_harness() {
-    size_t r = nondet_size_t();
-    size_t __r = nondet_size_t();
-    __CPROVER_assume(r == __r);
-
-    if (nondet_int()) {
+    if (nondet_bool()) {
```

Comment:
It would be just as easy to let CBMC do this naturally by declaring an uninitialized local variable, let CBMC give is a nondet value, and use that value in the conditional.  This has the added value that the value of the conditional (the value assigned to the local) will appear explicitly in any trace as the assignment to the local.